tests/snappi_tests/multidut/ecn/test_multidut_ecn_marking_with_snappi.py

```diff
@@ -0,0 +1,250 @@
+                      "The testbed name from testbed file doesn't match with MULTIDUT_TESTBED in variables.py ")
+        pytest_assert(len(snappi_port_list) >= tx_port_count + rx_port_count,
+                      "Need Minimum of {} ports defined in ansible/files/*links.csv file".
+                      format(tx_port_count + rx_port_count))
+
+        pytest_assert(len(rdma_ports['tx_ports']) >= tx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have the required Tx ports defined for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(MULTIDUT_TESTBED, testbed_subtype))
+
+        pytest_assert(len(rdma_ports['rx_ports']) >= rx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have the required Rx ports defined for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(MULTIDUT_TESTBED, testbed_subtype))
+
+        # Collect port names from rx_ports and tx_ports into a set for uniqueness
+        all_ports_set = set(port['port_name'] for port in rdma_ports['rx_ports'] + rdma_ports['tx_ports'])
+        pytest_assert(len(all_ports_set) >= tx_port_count + rx_port_count,
+                      'MULTIDUT_PORT_INFO doesn\'t have at least {} unique ports for \
+                      testbed {}, subtype {} in variables.py'.
+                      format(tx_port_count + rx_port_count, MULTIDUT_TESTBED, testbed_subtype))
+
+        logger.info('Running test for testbed subtype: {}'.format(testbed_subtype))
+
+        if is_snappi_multidut(duthosts):
+            snappi_ports = get_snappi_ports_for_rdma(snappi_port_list, rdma_ports,
+                                                     tx_port_count, rx_port_count, MULTIDUT_TESTBED)
+        else:
+            snappi_ports = snappi_port_list
+
+        testbed_config, port_config_list, snappi_ports = snappi_dut_base_config(duthosts,
+                                                                                snappi_ports,
+                                                                                snappi_api)
+
+    logger.info("Snappi Ports : {}".format(snappi_ports))
+    snappi_extra_params = SnappiTestParams()
+    snappi_extra_params.multi_dut_params.multi_dut_ports = snappi_ports
+
+    try:
+        run_ecn_marking_port_toggle_test(
+                                api=snappi_api,
+                                testbed_config=testbed_config,
+                                port_config_list=port_config_list,
+                                dut_port=snappi_ports[0]['peer_port'],
+                                test_prio_list=lossless_prio_list,
+                                prio_dscp_map=prio_dscp_map,
+                                snappi_extra_params=snappi_extra_params)
+    finally:
+        cleanup_config(duthosts, snappi_ports)
+
+
+test_flow_percent_list = [[90, 15], [53, 49], [15, 90], [49, 49], [50, 50]]
```

Comment:
how to read this ?

Reply:
This is the flow per cent rate for lossless TC 3 and 4.  so 90,15 means TC3 pkt injection at 90% of BW and TC4 at 15%. Combined more than 100% it will be.